docs/types/literal-suffixes.rst

```diff
@@ -0,0 +1,243 @@
+
+    suffix(42);
+    suffix(123, 2);
+    suffix("abc");
+    suffix(hex"12ff");
+    suffix(true);
+
+Note that the fractional decomposition is not performed for this kind of call - two-argument
+suffix functions must be explicitly called with two arguments.
+
+Regardless of the call syntax used and in contrast to applying a denomination, the result of the
+call is itself not considered a literal.
+As a consequence, it cannot be used as input of another suffix call, and calculations on it are performed
+within its type rather than in arbitrary precision (as is the case with calculations on rational number
+literals).
+
+.. note::
+    Function call syntax is the only way to pass a negative integer value into a suffix function
+    that has a parameter of a signed integer type.
+
+.. note::
+    As all free functions, suffix definitions can be :ref:`overloaded<overload-function>`.
+    Overloaded suffixes, however, cannot be invoked using the suffix call syntax.
+
+.. index:: ! fractional decomposition
+.. _fractional_decomposition:
+
+Fractional Decomposition
+------------------------
+
+To allow defining suffixes that work with fractional literals, like ``1.23``, the language allows
+a special form of a suffix definition.
+Such a suffix can be considered a more general form of a suffix taking a single integer argument.
+
+A single-parameter suffix can be applied only to those rational number literals which represent
+integers.
+Let us consider the following suffix definition:
+
+.. code-block:: solidity
+
+    function kg(uint grams) pure suffix returns (uint) {
+        return 1000 * grams;
+    }
+
+The ``kg`` suffix can receive integer values like ``123 kg``, ``1.23e2 kg``, or ``12300e-2 kg``.
+However, invoking such a suffix with a fractional number (e.g., ``1.23 kg``) triggers an error.
+We can fix that by adding an *exponent* parameter:
+
+.. code-block:: solidity
+
+    function kg(uint mantissa, uint exponent) pure suffix returns (uint) {
+        return 1000 * mantissa / 10**exponent;
+    }
+
+When defined this way, the suffix can handle all the literals it could previously, while ``1.23 kg``
+also becomes a valid expression, equivalent to ``kg(123, 2)``.
+
+More generally, the argument of such a suffix call is decomposed into two integer values (``mantissa``
+and ``exponent``), such that:
+
+#. ``mantissa * 10**-exponent`` is equal to the value of the literal.
+#. ``exponent`` is the smallest possible non-negative integer value satisfying the equation.
+
+The two rules provide unambiguous decomposition in all cases.
+For example:
+
+- ``123000`` is decomposed into ``123000 * 10**-0`` (i.e. ``123000`` for ``mantissa`` and ``0`` for ``exponent``).
+  Not ``123 * 10**3`` or ``123000000 * 10**-3``.
+
+  In general, when the suffix is invoked on an integer, ``mantissa`` is always equal to that integer
+  and ``exponent`` is ``0``.
+- ``1.23`` is decomposed into ``123 * 10**-2``, not ``1.23 * 10**-0`` or ``123000 * 10**-5``.
```

Comment:
but `-1.24` would work here too, right?

Reply:
No, the suffix would get `1.24` and `-` would only be applied to the result. See https://github.com/ethereum/solidity/pull/14121#discussion_r1169015325